packages/commonwealth/server/routes/topics/create_topic_handler.ts

```diff
@@ -13,7 +19,28 @@ export const createTopicHandler = async (
 ) => {
   const { user, chain, body } = req;
 
-  const topic = await controllers.topics.createTopic({ user, chain, body });
+  const validationSchema = z.object({
+    name: z.string().optional(),
+    description: z.string().optional(),
+    token_threshold: z.string().optional(),
+    featured_in_sidebar: z.coerce.boolean().optional(),
+    featured_in_new_post: z.coerce.boolean().optional(),
+    default_offchain_template: z.string().optional(),
+    chain_id: z.string().optional(),
+  });
+
+  const validationResult = validationSchema.safeParse(body);
+  if (validationResult.success === false) {
```

Comment:
Can be simplified to !validationResult.success

Reply:
(Unsure about this so feel free to push back) I feel like we can move the error validation layer into a middleware that takes in a schema (since it will be common for all routes). Then we would decouple our validation from our business logic, which would make it easier to read/test.

Reply:
> Can be simplified to !validationResult.success

Due to how the discriminated union works, you have to use `===` here to properly extract the `error` value. Using shorthand `!` syntax won't work

Reply:
Good catch. Good to note that we don't have strictNullChecks: true in our tsconfig, so we will have to manually handle the undefined/null cases instead of getting type checker to do it.

But this is a good workaround to discriminating the error type by handling these cases.

Reply:
Important to know that zod requires strict tsconfig, or we will find some nasty side effects.

https://zod.dev/?id=requirements